tests/test_search_aggregates.py

```diff
@@ -97,31 +98,83 @@ def generate_addon(addon_id, name, version):
 ]
 
 
+search_type = ArrayType(StructType([
+    StructField('engine', StringType(), False),
+    StructField('source', StringType(), False),
+    StructField('count', LongType(), False),
+]))
+
+
+main_summary_schema = [
+    ('client_id', 'a', StringType(), False),
+    ('submission_date', '20170101', StringType(), False),
+    ('os', 'windows', StringType(), True),
+    ('channel', 'release', StringType(), True),
+    ('country', 'DE', StringType(), True),
+    ('locale', 'de', StringType(), True),
+    ('search_cohort', None, StringType(), True),
+    ('app_version', '54.0.1', StringType(), True),
+    ('distribution_id', None, StringType(), True),
+    ('subsession_counter', 1, LongType(), True),
+    ('search_counts', [generate_search_count()], search_type, True),
+    ('active_addons', active_addons, addons_type, True),
+    # 30 minutes in active_ticks (30min * 60sec/min / 5sec/tick)
+    ('active_ticks', 360, LongType(), True),
+    (
+        'scalar_parent_browser_engagement_tab_open_event_count',
+        5,
+        LongType(),
+        True
+    ),
+    (
+        'scalar_parent_browser_engagement_max_concurrent_tab_count',
+        10,
+        LongType(),
+        True
+    ),
+    ('subsession_start_date', '2017-01-01 10:00', StringType(), False),
+    # One hour per ping
+    ('subsession_length', 60 * 60, LongType(), False),
+    # Roughly 2016-01-01
+    ('profile_creation_date', 16801, LongType(), False),
+    ('default_search_engine', 'google', StringType(), False),
+    (
+        'default_search_engine_data_load_path',
+        'jar:[app]/omni.ja!browser/google.xml',
+        StringType(),
+        False
+    ),
+    (
+        'default_search_engine_data_submission_url',
+        'https://www.google.com/search?q=&ie=utf-8&oe=utf-8&client=firefox-b',
+        StringType(),
+        False
+    ),
+]
+
+exploded_schema = filter(lambda x: x[0] != 'search_counts', main_summary_schema) + [
+    ('engine', 'google', StringType(), False),
+    ('source', 'urlbar', StringType(), False),
+    ('count', 4, LongType(), False),
+]
+
+derived_schema = exploded_schema + [
+    ('type', 'chrome-sap', StringType(), False),
+    ('addon_version', '0.9.5', StringType(), False),
+]
+
+
 @pytest.fixture()
 def generate_main_summary_data(define_dataframe_factory):
```

Comment:
I should have mentioned this earlier, but I don't think it really does much good to create a fixture that's basically just returning a function, since it just adds another layer of abstraction to unravel. I'd prefer if we just called these functions directly from either the test or a top-level fixture, as appropriate.

I can address this issue in a follow-up PR if you agree.

Reply:
That sounds good to me, but let's do it in a follow-up PR.